### ISSUE TITLE
I cannot run RustPython using wasmer

### ISSUE THREAD

**Issue by rafaelcaricio** [wasmerio/wasmer#1088]
### Summary

I seem to not be able to run RustPython with wasmer. I already looked into the issue https://github.com/RustPython/RustPython/issues/1637 but now it seems like wasmer cannot identify it uses `wasi_snapshot_preview1`. 

### Additional details

I cloned wasmer repo and ran: `make release` (latest commit in my local version is 0daaa75a7 ).
I also tried to run with the official released version of wasmer 0.11.0 without success, same error.

```
$ ./target/release/wasmer run ../RustPython/target/wasm32-wasi/release/rustpython.wasm
Error: Can't instantiate module: LinkError([ImportNotFound { namespace: "wasi_snapshot_preview1", name: "proc_exit" }, ImportNotFound { namespace: "wasi_unstable", name: "random_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "args_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "args_sizes_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "clock_time_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_close" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_read" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_write" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "path_filestat_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "path_open" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "poll_oneoff" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_prestat_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_prestat_dir_name" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "environ_sizes_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "environ_get" }])
```

Any ideas?

**Comment by rafaelcaricio** [comment]
I could run the released version of RustPython using my locally compiled version of wasmer. So the issue doesn't seem to come from wasmer then. I believe it might be related to the way I am compiling RustPython.

**Comment by MarkMcCaskey** [comment]
Hmm, it looks like it probably depends on which version of the WASI toolchain you're using.  We _should_ support both and it's a bug if we don't.  Can you check this with wasmer 0.12.0 and see if the problem is still there?  If it is, I'll fix it and make a 0.12.1 release today

**Comment by rafaelcaricio** [comment]
@MarkMcCaskey The error does happen with `wasmer` 0.12.0

```
$ wasmer run ../RustPython/target/wasm32-wasi/release/rustpython.wasm -- -c "print('hello')" 
Error: Can't instantiate module: LinkError([ImportNotFound { namespace: "wasi_snapshot_preview1", name: "proc_exit" }, ImportNotFound { namespace: "wasi_unstable", name: "random_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "args_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "args_sizes_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "clock_time_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_close" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_read" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_write" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "path_filestat_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "path_open" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "poll_oneoff" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_prestat_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "fd_prestat_dir_name" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "environ_sizes_get" }, ImportNotFound { namespace: "wasi_snapshot_preview1", name: "environ_get" }])
$ wasmer --version
wasmer 0.12.0
```
To reproduce yourself you can clone [`RustPython`](https://github.com/RustPython/RustPython) and compile with Rust nightly. You will need to change the `vm/Cargo.toml` file for this version of chrono `chrono = { git = "https://github.com/chronotope/chrono" }` Then compile:
```
$ cargo +nightly build --release --target wasm32-wasi --features="freeze-stdlib"
```

**Comment by MarkMcCaskey** [comment]
Thanks!  That's super useful,

I investigated it some more and found the issue: RustPython is compiling with an invalid set of imports.

The issue is that it's mixing `wasi_unstable` and `wasi_snapshot_preview1` imports.  This is due to the `getrandom` crate using the `wasi` crate which directly requires a specific version of WASI imports but most wasi imports come from the version of the Rust toolchain used to compile it.

I filed an issue on the `wasi` crate https://github.com/bytecodealliance/wasi/issues/37 about it.  Seems like it might not be an easy fix -- though I'm unfamiliar with the nightly release process and some of the inner magic of the rust stdilb, so I really have no idea.
